### ISSUE TITLE
Pregeneration idling

### ISSUE THREAD

**Issue by ArMiN231** [VolmitSoftware/Iris#182]
**Describe the bug**
The Pregeneration starts idling for no apparent reason. No further progress in the pregen. The ChunksMS is rising and nothing else happening in either the console or gui (apart from the ETA and Elapsed).

**To Reproduce**
1. Setup a new Server
2. Start the paper.jar
3. Join the game
4. Start the pregen with "/iris pregen 5k"
5. After some time the pregen starts idling and the ms are rising

**Expected behavior**
Generate the World till the configuered size is generated

**Server and Plugin Informations**
 - Installed plugins: Iris
 - Iris Version: V1.1.1
 - Server Platform and Version [eg: PaperSpigot 1.16.3 #240]: Paper #288
 - Operating System (if applicable): Windows 10

**Comment by jfrelik** [comment]
Same issue here on 1.16.4 Paper VolmitSoftware/Iris#288 and Paper VolmitSoftware/Iris#282
Using latest java flags from aikar
Ubuntu 18.04

Used pregen and left it for the night
https://del.dog/stygrevomy

**Comment by ghost** [comment]
> Same issue here on 1.16.4 Paper VolmitSoftware/Iris#288 and Paper VolmitSoftware/Iris#282
> Using latest java flags from aikar
> Ubuntu 18.04
> 
> Used pregen and left it for the night
> https://del.dog/stygrevomy

How much ram do you have? Currently running a test to reproduce the problem.

Edit: The issue could be associated to ram. The test pregen is still running.

**Comment by jfrelik** [comment]
10GB with 6 cpu cores, ssd disk

**Comment by ghost** [comment]
> 10GB with 6 cpu cores, ssd disk

Do you pregen on your pc or on your server through a panel or some serversoftware?

Edit: Pregen idling didn't happen on my pc when i used a bat to start my paper.jar 

"@echo off
title Server Console
java -Xmx4G -jar paper-288.jar
PAUSE"

I try to reproduce the idling to see if the bat do indeed fix the issue.

**Comment by jfrelik** [comment]
Pregen runs on my server via .sh script via screen

#/bin/bash

screen -d -m -S "mc_spigot_server" java -Xms10G -Xmx10G -XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:MaxGCPauseMillis=200 -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:+AlwaysPreTouch -XX:G1NewSizePercent=30 -XX:G1MaxNewSizePercent=40 -XX:G1HeapRegionSize=8M -XX:G1ReservePercent=20 -XX:G1HeapWastePercent=5 -XX:G1MixedGCCountTarget=4 -XX:InitiatingHeapOccupancyPercent=15 -XX:G1MixedGCLiveThresholdPercent=90 -XX:G1RSetUpdatingPauseTimePercent=5 -XX:SurvivorRatio=32 -XX:+PerfDisableSharedMem -XX:MaxTenuringThreshold=1 -Dusing.aikars.flags=https://mcflags.emc.gs -Daikars.new.flags=true -jar paper.jar nogui

**Comment by svdgoor** [comment]
Bump btw.

**Comment by cyberpwnn** [comment]
Please post your paper.yml and iris settings.json, having trouble causing the problem on my end. Also what your cpu is (core count is all i need if you have it)

**Comment by hydridity** [comment]
In my case, up until now, i've tested pregeneration on 2 different CPUs all attempts with default config didn't pass 35%, and all test iterations were done on pure new server (every time purging the old server and newly downloading paper and iris of the same version) **hence default newly generated content of paper.yml and iris config**
All tests were pregenerated with `/iris pregen 5000`
Iris: Iris-1.1.2
Paper: git-Paper-312 (MC: 1.16.4)
<details>
  <summary>Intel CPU</summary>

```
OS: Windows Version 10.0.18362.239
CPU: Intel(R) Core(TM) i5-3570 CPU @ 3.40GHz
Java version "1.8.0_221"
Java(TM) SE Runtime Environment (build 1.8.0_221-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.221-b11, mixed mode)
Total system ram: 16GB
Server assigned ram: 8GB
Launching server with: java -Xms3G -Xmx8G -jar custom.jar -nogui

```
</details>

<details>
  <summary>AMD CPU</summary>

```
OS: Ubuntu 20.04.1 LTS
CPU: AMD FX(tm)-6350 Six-Core Processor
openjdk version "11.0.9.1" 2020-11-04
OpenJDK Runtime Environment (build 11.0.9.1+1-Ubuntu-0ubuntu1.20.04)
OpenJDK 64-Bit Server VM (build 11.0.9.1+1-Ubuntu-0ubuntu1.20.04, mixed mode, sharing)
Total system ram: 20GB
Server assigned ram: 8GB or 12GB
Launching server with: java -Xmx8G -jar custom.jar -nogui    or    java -Xmx12G -jar custom.jar -nogui

```
</details>

However i was able to achieve 99% pregeneration on AMD setup, after modifying `"maxAsyncChunkPregenThreads"` value to match with the number of CPU cores
`"maxAsyncChunkPregenThreads": 6`
All other settings were left on default
The map was missing last part of the generational grid

While the pregeneration enters the "stuck phase" the cpu load will drop and does nothing at all, (no cpu load)
also the server is unable to stop itself properly via stop command or other stop signals except hard process kill, and will stuck on map saving stage with no CPU load or disk I/O at all, probably unable to cancel iris pregeneration which should be able to.

I hope the gathered information and problem behaviour will help somehow.

**Comment by ghost** [comment]
He fixed the bug today. We will get the fix with the next update.
Will be fixed in 1.1.5
